lib/registry.js

```diff
@@ -14,9 +13,62 @@ var BaseElement;
  */
 
 var ElementRegistry = createClass({
-  constructor: function() {
+  constructor: function(noDefault) {
     this.elementMap = {};
     this.elementDetection = [];
+    this.BaseElement = require('./primitives/base-element')(this);
```

Comment:
Yeah, my goal here was twofold:

1. Make is dead simple to access the base element via `registry.BaseElement`, e.g. for subclassing.
2. Make it so you can override the base element before loading other elements.

Unfortunately, we can't just add stuff to the base element after the fact because that's not how inheritance in the js world works. We would need to modify the base element, then create all the other elements because prototypal inheritance is essentially just cloning the prototype at the time of class definition. Changes to the base prototype aren't reflected in the new class after the class has been defined. This is especially problematic for things like source maps, where every single element should be updated and I don't have a good solution for it. A namespace module may need to iterate over all currently registered elements and adjust their prototypes accordingly.